tensorflow/python/ops/array_ops.py

```diff
@@ -3532,7 +3546,190 @@ def _get_paddings_constant(paddings):
     return [_get_paddings_constant(x) for x in paddings]
   else:
     return paddings
+  
+def _wrap_pad(tensor, paddings, name=None):
+  """Helper to create wrap pad option with ops already built.
 
+  See `pad` for more info on how this function works.
+  """
+  # use constant padding to create a tensor with final shape and original values.
+  padded = gen_array_ops.pad(tensor, paddings, name=name)
+  original_area_slice = []
+  for index, dim in enumerate(padded.TensorShape):
+    original_area_slice += dim - paddings[index,1] - tensor.dim_size(index)
+  pad_width = _as_pairs(pad_width, tensor.ndim)
+  axes = range(padded.ndims)
+  for axis, (left_index, right_index) in zip(axes, pad_width):
+            roi = _view_roi(padded, original_area_slice, axis)
```

Comment:
Could you instead just use `array_ops.tensor_strided_slice_update`?

Reply:
Thanks for the hint, I've used this instead. The approach is similar than before but just uses less helper functions.